ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -265,19 +258,29 @@
     }
     List<ReplicaId> backupReplicasToCheck = new ArrayList<>(backupReplicas);
     List<ReplicaId> downReplicasToCheck = new ArrayList<>(downReplicas);
-    if (includeNonOriginatingDcReplicas || this.originatingDcName == null) {
-      backupReplicas.forEach(this::addToEndOfPool);
+
+    // Add replicas that are neither in local dc nor in originating dc.
+    backupReplicas.forEach(this::addToEndOfPool);
```

Comment:
Right, with this change, we allows GET request to also try on non-originating replicas. Since these replicas will be tried with lower priority and we have `failOnNotFound` logic to terminate operation before exhausting all replicas.

Reply:
(The intention of this change is to simplify some logic in operation tracker. We already have many configs to control its behavior and some of them may not be needed anymore. )

Reply:
Ah that makes sense that `hasFailedOnNotFound()` will catch this first. I forgot about that. It is good to simplify the configuration as much as possible.